src/arrayMethodSort.js

```diff
@@ -4,8 +4,23 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (
+    compareFunction = (a, b) => a.toString() > b.toString(),
+  ) {
+    for (let i = 1; i < this.length; i++) {
+      const current = this[i];
+
+      let j = i - 1;
+
+      while (j >= 0 && compareFunction(this[j], current) > 0) {
+        this[j + 1] = this[j];
+        j--;
+      }
+
+      this[j + 1] = current;
+    }
```

Comment:
The provided sorting algorithm is an implementation of insertion sort, which is correct. However, make sure to handle cases where 'compareFunction' might not return a consistent result when 'a' and 'b' are equal or when 'compareFunction' is not a pure function.